packages/core/src/createGetBalances.ts

```diff
@@ -44,22 +46,40 @@ export const createGetBalances = ({ send }: Provider) =>
      * - `INVALID_THRESHOLD`: Invalid `threshold`
      * - Fetch error
      */
-    (addresses: ReadonlyArray<Hash>, threshold: number, callback?: Callback<Balances>): Promise<Balances> =>
-        Promise.resolve(
+    (
+        addresses: ReadonlyArray<Hash>,
+        threshold: number,
+        tips?: Hash[],
+        callback?: Callback<Balances>
+    ): Promise<Balances> => {
+        // If no tips are provided, switch arguments
+        if (tips && typeof tips === 'function') {
+            callback = tips
+            tips = []
+        }
+
+        return Promise.resolve(
             validate(
-                arrayValidator(hashValidator)(addresses, errors.INVALID_ADDRESS),
-                getBalancesThresholdValidator(threshold)
+                ...[
```

Comment:
I don't think we need to use spread here.

Reply:
It's needed here since [validate](https://github.com/iotaledger/iota.js/blob/next/packages/guards.ts#L186) accepts _any_ number of parameters.